pkg/transformers/greenmask/greenmask_integer_transformer_test.go

```diff
@@ -0,0 +1,415 @@
+	tests := []struct {
+		name        string
+		generator   transformers.GeneratorType
+		params      transformers.Parameters
+		wantErr     bool
+		errContains string
+	}{
+		{
+			name:      "ok - valid default parameters",
+			generator: transformers.Random,
+			params:    transformers.Parameters{},
+			wantErr:   false,
+		},
+		{
+			name:      "ok - valid custom parameters",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"size":      4,
+				"min_value": int64(-100),
+				"max_value": int64(100),
+			},
+			wantErr: false,
+		},
+		{
+			name:      "error - invalid size, too small",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"size": 0,
+			},
+			wantErr:     true,
+			errContains: "size must be greater than 0",
+		},
+		{
+			name:      "error - invalid size, too large",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"size": 9,
+			},
+			wantErr:     true,
+			errContains: "size must be less than or equal to 8",
+		},
+		{
+			name:      "error - wrong limits",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"min_value": int64(100),
+				"max_value": int64(99),
+			},
+			wantErr:     true,
+			errContains: "wrong limits",
+		},
+		{
+			name:      "error - invalid size type",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"size": "invalid",
+			},
+			wantErr:     true,
+			errContains: "size must be an integer",
+		},
+		{
+			name:      "error - invalid min_value type",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"min_value": "invalid",
+			},
+			wantErr:     true,
+			errContains: "min_value must be an integer",
+		},
+		{
+			name:      "error - invalid max_value type",
+			generator: transformers.Random,
+			params: transformers.Parameters{
+				"max_value": "invalid",
+			},
+			wantErr:     true,
+			errContains: "max_value must be an integer",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			transformer, err := NewIntegerTransformer(tt.generator, tt.params)
+			if tt.wantErr {
+				if err == nil {
+					t.Errorf("NewIntegerTransformer() expected error containing %v, got nil", tt.errContains)
+					return
+				}
+				if tt.errContains != "" && !strings.Contains(err.Error(), tt.errContains) {
+					t.Errorf("NewIntegerTransformer() error = %v, want error containing %v", err, tt.errContains)
+				}
+				return
+			}
+			if err != nil {
+				t.Errorf("NewIntegerTransformer() unexpected error = %v", err)
+				return
+			}
+			if transformer == nil {
+				t.Error("NewIntegerTransformer() returned nil transformer")
+			}
+		})
+	}
+}
+
+func TestIntegerTransformer_Transform_Random(t *testing.T) {
+	transformer, err := NewIntegerTransformer(transformers.Random, transformers.Parameters{
+		"size":      8,
+		"min_value": int64(-100),
+		"max_value": int64(100),
+	})
+	if err != nil {
```

Comment:
We use the [`testify`](https://github.com/stretchr/testify) go library, which simplifies all these very verbose test validations. You can use `require`, there are some examples here:
https://github.com/xataio/pgstream/blob/main/pkg/transformers/greenmask/greenmask_string_transformer_test.go